src/main/kotlin/uk/gov/justice/digital/hmpps/makerecalldecisionapi/service/RiskService.kt

```diff
@@ -62,6 +77,70 @@ class RiskService(
     }
   }
 
+  private suspend fun fetchCurrentScores(crn: String): Scores {
+    val currentScoresResponse = try {
+      getValue(arnApiClient.getCurrentScores(crn))!!
+    } catch (e: WebClientResponseException.NotFound) {
+      log.info("No cuurent scores available for CRN: $crn - ${e.message}")
+      listOf(
+        CurrentScoreResponse(
+          completedDate = "",
+          generalPredictorScore = GeneralPredictorScore(ogpStaticWeightedScore = "", ogpDynamicWeightedScore = "", ogpTotalWeightedScore = "", ogpRisk = ""),
+          riskOfSeriousRecidivismScore = RiskOfSeriousRecidivismScore(percentageScore = "", scoreLevel = ""),
+          sexualPredictorScore = SexualPredictorScore(ospIndecentPercentageScore = "", ospContactPercentageScore = "", ospIndecentScoreLevel = "", ospContactScoreLevel = "")
+        )
+      )
+    }
+    val latestScores = currentScoresResponse.maxByOrNull { LocalDateTime.parse(it.completedDate) }
+    val rsr = latestScores?.riskOfSeriousRecidivismScore
+    val osp = latestScores?.sexualPredictorScore
+    val osg = latestScores?.generalPredictorScore
+    return Scores(
+      rsr = RSR(level = rsr?.scoreLevel ?: "", score = rsr?.percentageScore ?: "", type = "RSR"),
+      ospc = OSPC(level = osp?.ospContactScoreLevel ?: "", score = osp?.ospContactPercentageScore ?: "", type = "OSP/C"),
+      ospi = OSPI(level = osp?.ospIndecentScoreLevel ?: "", score = osp?.ospIndecentPercentageScore ?: "", type = "OSP/I"),
+      ogrs = OGRS(level = osg?.ogpRisk ?: "", score = osg?.ogpTotalWeightedScore, type = "OGRS") // TODO check if 'total' is correct field
+    )
+  }
+
+  private suspend fun fetchHistoricalScores(crn: String): List<HistoricalScore> {
+    val historicalScoresResponse = try {
+      getValue(arnApiClient.getHistoricalScores(crn))!!
+    } catch (e: WebClientResponseException.NotFound) {
+      log.info("No historical scores available for CRN: $crn - ${e.message}")
+      listOf(
+        HistoricalScoreResponse(
+          rsrPercentageScore = "",
+          rsrScoreLevel = "",
+          ospcPercentageScore = "",
+          ospcScoreLevel = "",
+          ospiPercentageScore = "",
+          ospiScoreLevel = "",
+          calculatedDate = null
+        )
+      )
+    }
+    return historicalScoresResponse
+      .map {
+        HistoricalScore(
+          date = it.calculatedDate?.let { it1 -> formatDateTimeStamp(it1) } ?: "",
+          scores = Scores(
+            rsr = RSR(level = it.rsrScoreLevel ?: "", score = it.rsrPercentageScore ?: "", type = "RSR"),
+            ospc = OSPC(level = it.ospcScoreLevel ?: "", score = it.ospcPercentageScore ?: "", type = "OSP/C"),
+            ospi = OSPI(level = it.ospiScoreLevel ?: "", score = it.ospiPercentageScore ?: "", type = "OSP/I"),
+            ogrs = OGRS(level = "", score = "", type = "OGRS") // TODO - discuss with ARN team
+          )
+        )
+      }
+  }
+
+  private fun formatDateTimeStamp(localDateTimeString: String): String {
```

Comment:
Maybe move this out to some helper utility class at some point so it can easily be reused by other services as required